aws/resource_aws_wafv2_web_acl_association.go

```diff
@@ -0,0 +1,133 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsWafv2WebACLAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2WebACLAssociationCreate,
+		Read:   resourceAwsWafv2WebACLAssociationRead,
+		Delete: resourceAwsWafv2WebACLAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				webAclArn, resourceArn, err := resourceAwsWafv2ACLAssociationDecodeId(d.Id())
+				if err != nil {
+					return nil, fmt.Errorf("Error reading resource ID: %s", err)
+				}
+				d.Set("resource_arn", resourceArn)
+				d.Set("web_acl_arn", webAclArn)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_arn": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"web_acl_arn": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsWafv2WebACLAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	resourceArn := d.Get("resource_arn").(string)
+	webAclArn := d.Get("web_acl_arn").(string)
+	params := &wafv2.AssociateWebACLInput{
+		ResourceArn: aws.String(resourceArn),
+		WebACLArn:   aws.String(webAclArn),
+	}
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
we could also emulate the pattern in the `wafv2_regex_pattern_set` PR of adding to the err handlers in `config.go` for this error type/operation combination

Reply:
@anGie44 are you sure you want to move this?
This is not really a generic error but more related to this resource so I'm not sure if we should move this, also according to what @bflad wrote here [here](https://github.com/terraform-providers/terraform-provider-aws/pull/12119#discussion_r419806963).
If you still want it to be moved,let me know and I'll move it to `config.go`

Reply:
ah sorry I glazed over the exception type here - yep you're totally right, it should stay here as it's specific to the resource